### PR TITLE
Change visual script round function output type to float

### LINKED ISSUE #37844: Round or Switch Bug
**Godot version:**
3.2.1 stable
**OS/device including version:**
Windows 10
**Issue description:**
For some unknown reason, it doesn't understand that when a float is rounded to an int it should be an int and not still a float. in the example below, i have a vector with (90,0) that i use round on the X value, plug that value in to the switch and if it is an int=90 it should print 90... however it acts like 90!=90 and doesn't print that last print.
**Steps to reproduce:**
use round on any float, plug the int in to a switch, make sure the values are equal on both the round you get out and the switch case you want to run
**Minimal reproduction project:**
[Bug.zip](https://github.com/godotengine/godot/files/4469253/Bug.zip)

### PR DESCRIPTION
**Pull request by nicomitchell**
Fixes #37844

### CODE REVIEW

**Comment by akien-mga** [comment]
Thanks!

File: modules/visual_script/visual_script_builtin_funcs.cpp [hunk 1]
```diff
@@ -508,7 +508,7 @@ PropertyInfo VisualScriptBuiltinFunc::get_output_value_port_info(int p_idx) cons
 		} break;
 		case MATH_POSMOD:
 		case MATH_ROUND: {
-			t = Variant::INT;
+			t = Variant::FLOAT;
```
Comment (bojidar-bg) [inline modules/visual_script/visual_script_builtin_funcs.cpp:511]: Unfortunately, MATH_POSMOD is now incorrect, since it does actually return int.

Comment (nicomitchell) [reply modules/visual_script/visual_script_builtin_funcs.cpp:511]: My mistake, forgot about the switch/case fallthrough! This should be addressed by pull #37890

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2. (Together with #37890.)

**Pull request merged**